### PR TITLE
Fix getting device number from AMD Kria devices

### PR DESCRIPTION
**Pull request by superm1**
On Kria devices the I2C string has extra components, so it can't
be taken as the last argument:

"/sys/devices/platform/axi/ff030000.i2c/i2c-1/1-0050"

So look for the first token that starts with "i2c-" instead.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-udev-device.c [hunk 1]
```diff
@@ -479,24 +479,21 @@ gboolean
...
-		if (!g_ascii_isdigit(path->str[i])) {
-			g_string_erase(path, 0, i + 1);
-			break;
-		}
-	}
-	if (path->len > 0) {
-		if (!fu_strtoull(path->str,
+	for (guint i = 0; tokens[i] != NULL; i++) {
+		if (!g_str_has_prefix(tokens[i], "i2c-"))
+			continue;
```
Comment (hughsie) [inline libfwupdplugin/fu-udev-device.c:487]: I don't think putting i2c here is a good idea. Can we handle this case in the FuI2cDevice file?

Comment (superm1) [reply libfwupdplugin/fu-udev-device.c]: fixed

File: libfwupdplugin/fu-i2c-device.c [hunk 2]
```diff
@@ -80,9 +80,33 @@ fu_i2c_device_probe(FuDevice *device, GError **error)
...
 	if (fu_udev_device_get_device_file(FU_UDEV_DEVICE(self)) == NULL) {
-		g_autofree gchar *devfile =
-		    g_strdup_printf("/dev/i2c-%u",
-				    (guint)fu_udev_device_get_number(FU_UDEV_DEVICE(self)));
+		const gchar *sysfs = fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(self));
+		g_auto(GStrv) tokens = fu_strsplit(sysfs, strlen(sysfs), "/", -1);
+		g_autofree gchar *devfile = NULL;
+		guint64 number = G_MAXUINT64;
+
+		for (guint i = 0; tokens[i] != NULL; i++) {
```
Comment (hughsie) [inline libfwupdplugin/fu-i2c-device.c:88]: isn't this going from root to tip? don't you want to go from tip to root?

Comment (superm1) [reply libfwupdplugin/fu-i2c-device.c:88]: Yeah; it's root to tip.  But I had figured for an I2C device you _want_ the root bus.  Disagree?

Comment (hughsie) [reply libfwupdplugin/fu-i2c-device.c:88]: Maybe it's an angels on pinheads thing. I don't mind too much each way. I've got a megacommit which adds fake sysfs devices coming up (to test all this fu-udev class stuff I added) -- we can add a test for this condition there.

**Review by hughsie — APPROVED** [review]

File: libfwupdplugin/fu-i2c-device.c (hunk 2 above)
Comment (superm1) [reply libfwupdplugin/fu-i2c-device.c:88]: > I've got a megacommit which adds fake sysfs devices coming up (to test all this fu-udev class stuff I added

Like we can actually get to the point someone can tar up /sys and we can emulate someone's udev devices?

Comment (hughsie) [reply libfwupdplugin/fu-i2c-device.c:88]: > Like we can actually get to the point someone can tar up /sys and we can emulate someone's udev devices?

Exactly that

**Pull request merged**